vowpalwabbit/parser.cc

```diff
@@ -147,17 +75,17 @@ bool is_test_only(uint32_t counter, uint32_t period, uint32_t after, bool holdou
 
 parser* new_parser()
 {
-  parser& ret = calloc_or_throw<parser>();
-  ret.input = new io_buf;
-  ret.output = new io_buf;
-  ret.local_example_number = 0;
-  ret.in_pass_counter = 0;
-  ret.ring_size = 1 << 8;
-  ret.done = false;
-  ret.used_index = 0;
-  ret.jsonp = nullptr;
-
-  return &ret;
+  auto ret = new parser();
+  ret->input = new io_buf;
```

Comment:
This doesn't matter at a macro scale, but it's slightly more elegant to auto& ret then use '.' to dereference instead of '->'.